src/gerbil/runtime/c3.ss

```diff
@@ -0,0 +1,108 @@
+;;; -*- Gerbil -*-
+;;; © fare
+;;; The C3 Linearization algorithm to topologically sort an inheritance DAG
+;;; into a precedence list such that:
+;;;  1. The precedence list is a linearization of the inheritance DAG
+;;;  2. Every precedence list includes as a sublist the precedence list of every super
+;;;  3. The list of supers is a sublist of the precedence list
+;;;  4. The elements that appear earlier in the list of precedence lists followed by
+;;;     the super list appear earlier in the precedence list.
+;;;     (This heuristic choice of an order makes the linearization deterministic).
+;;; In the above, sublist means that the super list includes all the sublist elements
+;;; in the same order but not necessarily consecutively.
+;;;
+;;; C3 has since been adopted for multiple inheritance by many modern object systems
+;;; for its ordering consistency properties:
+;;; OpenDylan, Python, Raku, Parrot, Solidity, PGF/TikZ.
+;;;       https://en.wikipedia.org/wiki/C3_linearization
+;;;       https://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.19.3910
+;;;
+;;; See tests in ../test/c3-test.ss
+
+prelude: :gerbil/core
+package: gerbil/runtime
+namespace: #f
+
+(export #t)
```

Comment:
do we need to export everything? I think we can get by with just c3-linearize, no?

Reply:
we should probably move the utilities to "util" and just keep the c3 stuff here.